### PR TITLE
Improve nonce handling and permissions in crop settings API

### PR DESCRIPTION
**Pull request by mtashjianjr-godaddy**
Enhance nonce validation and permission checks for crop settings, ensuring only authorized users can access and modify images. Additionally, conditionally include nonce values in localized script data.

### CODE REVIEW

File: includes/admin/class-coblocks-crop-settings.php [hunk 1]
```diff
@@ -86,26 +86,22 @@ public function hide_cropped_from_library( $query ) {
 	 * Retrieve the original image.
 	 */
 	public function get_original_image() {
-		$nonce = filter_input( INPUT_POST, 'nonce' );
-
-		if ( ! $nonce ) {
-
-			wp_send_json_error( 'No nonce value present.' );
-
+		if ( ! wp_verify_nonce( filter_input( INPUT_POST, 'nonce' ), 'cropSettingsOriginalImageNonce' ) ) {
```
Comment (eherman-godaddy) [inline includes/admin/class-coblocks-crop-settings.php:89]: I'm wondering if we should wrap this filter_input with a `sanitize_text_field()` just to ensure it's sanitized before checking. This has been getting flagged in plugin reviews, and the security team might flag it if we don't.

```php
if ( ! wp_verify_nonce( sanitize_text_field( filter_input( INPUT_POST, 'nonce' ) ), 'cropSettingsOriginalImageNonce' ) ) {
```

File: includes/admin/class-coblocks-crop-settings.php [hunk 2]
```diff
@@ -127,18 +123,22 @@ public function get_original_image() {
 	 * Cropping.
 	 */
 	public function api_crop() {
-		$nonce = filter_input( INPUT_POST, 'nonce' );
-
-		if ( ! $nonce ) {
-
-			wp_send_json_error( 'No nonce value present.' );
+		if ( ! wp_verify_nonce( filter_input( INPUT_POST, 'nonce' ), 'cropSettingsNonce' ) ) {
```
Comment (eherman-godaddy) [inline includes/admin/class-coblocks-crop-settings.php:126]: `sanitize_text_field` here as well

File: includes/admin/class-coblocks-crop-settings.php (hunk 1 above)
Comment (mtashjianjr-godaddy) [reply includes/admin/class-coblocks-crop-settings.php:89]: That's odd, considering it's not stored in a database or rendered on the website. We can add it regardless.

Comment (eherman-godaddy) [reply includes/admin/class-coblocks-crop-settings.php]: Yeah, I agree. It throws a warning in our plugin scanner tool and in the plugin check plugin.

**Pull request merged**